src/Backends/InterfaceSettings.vala

```diff
@@ -24,22 +24,35 @@ public class SettingsDaemon.Backends.InterfaceSettings : GLib.Object {
     private const string CURSOR_SIZE = "cursor-size";
     private const string LOCATE_POINTER = "locate-pointer";
     private const string TEXT_SCALING_FACTOR = "text-scaling-factor";
+
+    private const string PICTURE_OPTIONS = "picture-options";
+    private const string PRIMARY_COLOR = "primary-color";
+    private const string PICTURE_URI = "picture-uri";
+    private const string PICTURE_URI_DARK = "picture-uri-dark";
+
     private const string DOCUMENT_FONT_NAME = "document-font-name";
     private const string FONT_NAME = "font-name";
     private const string MONOSPACE_FONT_NAME = "monospace-font-name";
 
     public unowned AccountsService accounts_service { get; construct; }
+    public unowned DisplayManager.AccountsService display_manager_accounts_service { get; construct; }
 
     private GLib.Settings interface_settings;
+    private GLib.Settings background_settings;
 
-    public InterfaceSettings (AccountsService accounts_service) {
-        Object (accounts_service: accounts_service);
+    public InterfaceSettings (AccountsService accounts_service, DisplayManager.AccountsService display_manager_accounts_service) {
+        Object (
+            accounts_service: accounts_service,
+            display_manager_accounts_service: display_manager_accounts_service
+        );
     }
 
     construct {
         interface_settings = new GLib.Settings ("org.gnome.desktop.interface");
+        background_settings = new GLib.Settings ("org.gnome.desktop.background");
 
         sync_gsettings_to_accountsservice ();
+        sync_background_to_greeter ();
```

Comment:
are this line necessary? i assume that the daemon will be runs for the lifetime of the session, and that it's not possible to change the background of one session from another session.